src/Utility/ExprFunctions.C

```diff
@@ -105,51 +105,31 @@ EvalFunc::EvalFunc (const char* function, const char* x, Real eps)
     expr.resize(nalloc);
     arg.resize(nalloc);
     for (size_t i = 0; i < nalloc; ++i) {
-      expr[i] = new Expression;
-      f[i] = new FunctionList;
-      v[i] = new ValueList;
+      expr[i] = std::make_unique<Expression>();
+      f[i] = std::make_unique<FunctionList>();
+      v[i] = std::make_unique<ValueList>();
       f[i]->AddDefaultFunctions();
       v[i]->AddDefaultValues();
       v[i]->Add(x,0.0,false);
-      expr[i]->SetFunctionList(f[i]);
-      expr[i]->SetValueList(v[i]);
+      expr[i]->SetFunctionList(f[i].get());
+      expr[i]->SetValueList(v[i].get());
       expr[i]->Parse(function);
       arg[i] = v[i]->GetAddress(x);
     }
   }
   catch (ExprEval::Exception& e) {
-    this->cleanup();
     ExprException(e,"parsing",function);
   }
 }
 
 
-EvalFunc::~EvalFunc ()
-{
-  this->cleanup();
-}
-
-
-void EvalFunc::cleanup ()
-{
-  for (Expression* it : expr)
-    delete it;
-  for (FunctionList* it : f)
-    delete it;
-  for (ValueList* it : v)
-    delete it;
-  delete gradient;
-  expr.clear();
-  f.clear();
-  v.clear();
-  arg.clear();
-}
+EvalFunc::~EvalFunc () = default;
```

Comment:
The now empty (or default) destructor should then be inlined in the header file (and the current destructor doxy's are no longer valid).

Reply:
if i put it in the header we need the definition of the types of the pointers being deleted in the header. which we do not currently have, and i prefer to keep the forwarding. as for the doxy it's still technically correct since we are running the destructors of the unique_ptr's, but i can rephrase it.

Reply:
OK, I was unaware of that. Then keep it in the C-file, we want to have the header-files as slim as possible wrt. includes.

And yes, the destructors will now still delete the heap-allocated items, but it is now done behind the scene (in uniq_ptr) and not in "our" code which the doxy is referring to. This can be said about many of the other "empty" destructors as well. I have used to write only what is done explicitly by it - not what is done via members or parent class implementation. But no big deal - kan leave it for now.

Reply:
it's the same as bare pointers, there is nothing magical about unique_ptrs in this respect.

i have rewritten to say defaulted constructors with a details tag to explain why they are not inlined.